onnxruntime/test/xctest/xcgtest.mm

```diff
@@ -0,0 +1,269 @@
+        NSString *description = @(test_part_result.message());
+        [_testCase recordFailureWithDescription:description
+                                         inFile:path
+                                         atLine:(lineNumber >= 0 ? (NSUInteger)lineNumber : 0)
+                                       expected:YES];
+    }
+
+private:
+    XCTestCase *_testCase;
+};
+
+/**
+ * Registers an XCTestCase subclass for each Google Test case.
+ *
+ * Generating these classes allows Google Test cases to be represented as peers
+ * of standard XCTest suites and supports filtering of test runs to specific
+ * Google Test cases or individual tests via Xcode.
+ */
+@interface GoogleTestLoader : NSObject
+@end
+
+/**
+ * Base class for the generated classes for Google Test cases.
+ */
+@interface GoogleTestCase : XCTestCase
+@end
+
+@implementation GoogleTestCase
+
+/**
+ * Associates generated Google Test classes with the test bundle.
+ *
+ * This affects how the generated test cases are represented in reports. By
+ * associating the generated classes with a test bundle the Google Test cases
+ * appear to be part of the same test bundle that this source file is compiled
+ * into. Without this association they appear to be part of a bundle
+ * representing the directory of an internal Xcode tool that runs the tests.
+ */
++ (NSBundle *)bundleForClass {
+    return [NSBundle bundleForClass:[GoogleTestLoader class]];
+}
+
+/**
+ * Implementation of +[XCTestCase testInvocations] that returns an array of test
+ * invocations for each test method in the class.
+ *
+ * This differs from the standard implementation of testInvocations, which only
+ * adds methods with a prefix of "test".
+ */
++ (NSArray *)testInvocations {
+    NSMutableArray *invocations = [NSMutableArray array];
+
+    unsigned int methodCount = 0;
+    Method *methods = class_copyMethodList([self class], &methodCount);
+
+    for (unsigned int i = 0; i < methodCount; i++) {
+        SEL sel = method_getName(methods[i]);
+        NSMethodSignature *sig = [self instanceMethodSignatureForSelector:sel];
+        NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:sig];
+        [invocation setSelector:sel];
+        [invocations addObject:invocation];
+    }
+
+    free(methods);
+
+    return invocations;
+}
+
+@end
+
+/**
+ * Runs a single test.
+ */
+static void RunTest(id self, SEL _cmd) {
+    XCTestListener *listener = new XCTestListener(self);
+    UnitTest *googleTest = UnitTest::GetInstance();
+    googleTest->listeners().Append(listener);
+
+    NSString *testKey = [NSString stringWithFormat:@"%@.%@", [self class], NSStringFromSelector(_cmd)];
+    NSString *testFilter = GoogleTestFilterMap[testKey];
+    XCTAssertNotNil(testFilter, @"No test filter found for test %@", testKey);
+
+    testing::GTEST_FLAG(filter) = [testFilter UTF8String];
+    ortenv_setup();
+    (void)RUN_ALL_TESTS();
```

Comment:
Seems this will run for every single test, not sure if it is possible to move it to GoogleTestLoader so it will only be called once?

Reply:
In the current implementation of this wrapper, the case will run one by one and global gtest initialization will run again and again as well. And all ort unit tests are ok with that, except data_type and have fixed it.

So it's hard to tell whether this env setup should run once or more. I can move it into global and see if everything works too.